cmake/sysbuild/fast_pair_hex.cmake

```
@@ -47,6 +47,24 @@ function(fast_pair_hex_pm)
     )
 endfunction()
 
+# This function takes partition nodelabel, extracts all numbers (hex addresses) preceded by @ and
+# adds them up to get the absolute partition address. The result is stored in the result variable.
+# This function should probably one day be moved to some common utilities file.
+function(get_partition_absolute_addr result partition_nodelabel)
+  # Use regex to match all hex numbers preceded by @
+  string(REGEX MATCHALL "@([0-9a-f]+)" matches "${partition_nodelabel}")
+
+  set(absolute_addr 0x0)
+
+  # Extract the addresses from the matches and add them
+  foreach(match IN LISTS matches)
+    string(REGEX REPLACE "@([0-9a-f]+)" "\\1" addr "${match}")
+    math(EXPR absolute_addr "${absolute_addr} + 0x${addr}" OUTPUT_FORMAT HEXADECIMAL)
```

Consider decoupling conversion from the sum operation:

```suggestion
    string(REGEX REPLACE "@([0-9a-f]+)" "0x\\1" addr "${match}")
    math(EXPR absolute_addr "${absolute_addr} + ${addr}" OUTPUT_FORMAT HEXADECIMAL)
```